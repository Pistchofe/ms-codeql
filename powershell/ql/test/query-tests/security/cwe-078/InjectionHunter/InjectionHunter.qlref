queries/security/cwe-078/InjectionHunter/UserInputToDangerousMethod.ql