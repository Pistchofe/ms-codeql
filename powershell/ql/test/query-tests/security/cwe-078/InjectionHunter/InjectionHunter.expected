edges
| test.ps1:3:11:3:20 | UserInput | test.ps1:4:23:4:52 | Get-Process -Name $UserInput | provenance |  |
| test.ps1:9:11:9:20 | UserInput | test.ps1:10:9:10:38 | Get-Process -Name $UserInput | provenance |  |
| test.ps1:15:11:15:20 | UserInput | test.ps1:16:50:16:79 | Get-Process -Name $UserInput | provenance |  |
| test.ps1:21:11:21:20 | UserInput | test.ps1:22:41:22:70 | Get-Process -Name $UserInput | provenance |  |
| test.ps1:21:11:21:20 | UserInput | test.ps1:22:60:22:69 | UserInput | provenance |  |
| test.ps1:27:11:27:20 | UserInput | test.ps1:28:38:28:67 | Get-Process -Name $UserInput | provenance |  |
| test.ps1:27:11:27:20 | UserInput | test.ps1:28:57:28:66 | UserInput | provenance |  |
| test.ps1:33:11:33:20 | UserInput | test.ps1:34:14:34:46 | public class Foo { $UserInput } | provenance |  |
| test.ps1:39:11:39:20 | UserInput | test.ps1:40:30:40:62 | public class Foo { $UserInput } | provenance |  |
| test.ps1:45:11:45:20 | UserInput | test.ps1:48:30:48:34 | code | provenance |  |
| test.ps1:73:11:73:20 | UserInput | test.ps1:75:25:75:54 | Get-Process -Name $UserInput | provenance |  |
| test.ps1:80:11:80:20 | UserInput | test.ps1:82:16:82:45 | Get-Process -Name $UserInput | provenance |  |
| test.ps1:87:11:87:20 | UserInput | test.ps1:89:12:89:28 | ping $UserInput | provenance |  |
| test.ps1:102:11:102:20 | UserInput | test.ps1:106:33:106:62 | Get-Process -Name $UserInput | provenance |  |
| test.ps1:112:11:112:20 | UserInput | test.ps1:116:58:116:87 | Get-Process -Name $UserInput | provenance |  |
| test.ps1:122:11:122:20 | UserInput | test.ps1:124:34:124:43 | UserInput | provenance |  |
| test.ps1:129:11:129:20 | UserInput | test.ps1:131:28:131:37 | UserInput | provenance |  |
| test.ps1:136:11:136:20 | UserInput | test.ps1:138:28:138:37 | UserInput | provenance |  |
| test.ps1:165:11:165:20 | UserInput | test.ps1:168:50:168:59 | UserInput | provenance |  |
| test.ps1:173:11:173:20 | UserInput | test.ps1:176:63:176:72 | UserInput | provenance |  |
| test.ps1:189:11:189:20 | UserInput | test.ps1:192:23:192:54 | Get-Process -Name "$escaped" | provenance |  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:197:46:197:51 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:198:46:198:51 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:199:46:199:51 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:200:46:200:51 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:201:46:201:51 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:202:46:202:51 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:203:46:203:51 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:204:46:204:51 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:206:48:206:53 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:207:48:207:53 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:208:48:208:53 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:210:41:210:46 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:211:41:211:46 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:212:36:212:41 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:213:36:213:41 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:214:36:214:41 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:218:42:218:47 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:219:42:219:47 | input | provenance | Src:MaD:11464  |
| test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:221:33:221:38 | input | provenance | Src:MaD:11464  |
| test.ps1:197:46:197:51 | input | test.ps1:3:11:3:20 | UserInput | provenance |  |
| test.ps1:198:46:198:51 | input | test.ps1:9:11:9:20 | UserInput | provenance |  |
| test.ps1:199:46:199:51 | input | test.ps1:15:11:15:20 | UserInput | provenance |  |
| test.ps1:200:46:200:51 | input | test.ps1:21:11:21:20 | UserInput | provenance |  |
| test.ps1:201:46:201:51 | input | test.ps1:27:11:27:20 | UserInput | provenance |  |
| test.ps1:202:46:202:51 | input | test.ps1:33:11:33:20 | UserInput | provenance |  |
| test.ps1:203:46:203:51 | input | test.ps1:39:11:39:20 | UserInput | provenance |  |
| test.ps1:204:46:204:51 | input | test.ps1:45:11:45:20 | UserInput | provenance |  |
| test.ps1:206:48:206:53 | input | test.ps1:73:11:73:20 | UserInput | provenance |  |
| test.ps1:207:48:207:53 | input | test.ps1:80:11:80:20 | UserInput | provenance |  |
| test.ps1:208:48:208:53 | input | test.ps1:87:11:87:20 | UserInput | provenance |  |
| test.ps1:210:41:210:46 | input | test.ps1:102:11:102:20 | UserInput | provenance |  |
| test.ps1:211:41:211:46 | input | test.ps1:112:11:112:20 | UserInput | provenance |  |
| test.ps1:212:36:212:41 | input | test.ps1:122:11:122:20 | UserInput | provenance |  |
| test.ps1:213:36:213:41 | input | test.ps1:129:11:129:20 | UserInput | provenance |  |
| test.ps1:214:36:214:41 | input | test.ps1:136:11:136:20 | UserInput | provenance |  |
| test.ps1:218:42:218:47 | input | test.ps1:165:11:165:20 | UserInput | provenance |  |
| test.ps1:219:42:219:47 | input | test.ps1:173:11:173:20 | UserInput | provenance |  |
| test.ps1:221:33:221:38 | input | test.ps1:189:11:189:20 | UserInput | provenance |  |
nodes
| test.ps1:3:11:3:20 | UserInput | semmle.label | UserInput |
| test.ps1:4:23:4:52 | Get-Process -Name $UserInput | semmle.label | Get-Process -Name $UserInput |
| test.ps1:9:11:9:20 | UserInput | semmle.label | UserInput |
| test.ps1:10:9:10:38 | Get-Process -Name $UserInput | semmle.label | Get-Process -Name $UserInput |
| test.ps1:15:11:15:20 | UserInput | semmle.label | UserInput |
| test.ps1:16:50:16:79 | Get-Process -Name $UserInput | semmle.label | Get-Process -Name $UserInput |
| test.ps1:21:11:21:20 | UserInput | semmle.label | UserInput |
| test.ps1:22:41:22:70 | Get-Process -Name $UserInput | semmle.label | Get-Process -Name $UserInput |
| test.ps1:22:60:22:69 | UserInput | semmle.label | UserInput |
| test.ps1:27:11:27:20 | UserInput | semmle.label | UserInput |
| test.ps1:28:38:28:67 | Get-Process -Name $UserInput | semmle.label | Get-Process -Name $UserInput |
| test.ps1:28:57:28:66 | UserInput | semmle.label | UserInput |
| test.ps1:33:11:33:20 | UserInput | semmle.label | UserInput |
| test.ps1:34:14:34:46 | public class Foo { $UserInput } | semmle.label | public class Foo { $UserInput } |
| test.ps1:39:11:39:20 | UserInput | semmle.label | UserInput |
| test.ps1:40:30:40:62 | public class Foo { $UserInput } | semmle.label | public class Foo { $UserInput } |
| test.ps1:45:11:45:20 | UserInput | semmle.label | UserInput |
| test.ps1:48:30:48:34 | code | semmle.label | code |
| test.ps1:73:11:73:20 | UserInput | semmle.label | UserInput |
| test.ps1:75:25:75:54 | Get-Process -Name $UserInput | semmle.label | Get-Process -Name $UserInput |
| test.ps1:80:11:80:20 | UserInput | semmle.label | UserInput |
| test.ps1:82:16:82:45 | Get-Process -Name $UserInput | semmle.label | Get-Process -Name $UserInput |
| test.ps1:87:11:87:20 | UserInput | semmle.label | UserInput |
| test.ps1:89:12:89:28 | ping $UserInput | semmle.label | ping $UserInput |
| test.ps1:102:11:102:20 | UserInput | semmle.label | UserInput |
| test.ps1:106:33:106:62 | Get-Process -Name $UserInput | semmle.label | Get-Process -Name $UserInput |
| test.ps1:112:11:112:20 | UserInput | semmle.label | UserInput |
| test.ps1:116:58:116:87 | Get-Process -Name $UserInput | semmle.label | Get-Process -Name $UserInput |
| test.ps1:122:11:122:20 | UserInput | semmle.label | UserInput |
| test.ps1:124:34:124:43 | UserInput | semmle.label | UserInput |
| test.ps1:129:11:129:20 | UserInput | semmle.label | UserInput |
| test.ps1:131:28:131:37 | UserInput | semmle.label | UserInput |
| test.ps1:136:11:136:20 | UserInput | semmle.label | UserInput |
| test.ps1:138:28:138:37 | UserInput | semmle.label | UserInput |
| test.ps1:165:11:165:20 | UserInput | semmle.label | UserInput |
| test.ps1:168:50:168:59 | UserInput | semmle.label | UserInput |
| test.ps1:173:11:173:20 | UserInput | semmle.label | UserInput |
| test.ps1:176:63:176:72 | UserInput | semmle.label | UserInput |
| test.ps1:189:11:189:20 | UserInput | semmle.label | UserInput |
| test.ps1:192:23:192:54 | Get-Process -Name "$escaped" | semmle.label | Get-Process -Name "$escaped" |
| test.ps1:195:10:195:32 | Call to Read-Host | semmle.label | Call to Read-Host |
| test.ps1:197:46:197:51 | input | semmle.label | input |
| test.ps1:198:46:198:51 | input | semmle.label | input |
| test.ps1:199:46:199:51 | input | semmle.label | input |
| test.ps1:200:46:200:51 | input | semmle.label | input |
| test.ps1:201:46:201:51 | input | semmle.label | input |
| test.ps1:202:46:202:51 | input | semmle.label | input |
| test.ps1:203:46:203:51 | input | semmle.label | input |
| test.ps1:204:46:204:51 | input | semmle.label | input |
| test.ps1:206:48:206:53 | input | semmle.label | input |
| test.ps1:207:48:207:53 | input | semmle.label | input |
| test.ps1:208:48:208:53 | input | semmle.label | input |
| test.ps1:210:41:210:46 | input | semmle.label | input |
| test.ps1:211:41:211:46 | input | semmle.label | input |
| test.ps1:212:36:212:41 | input | semmle.label | input |
| test.ps1:213:36:213:41 | input | semmle.label | input |
| test.ps1:214:36:214:41 | input | semmle.label | input |
| test.ps1:218:42:218:47 | input | semmle.label | input |
| test.ps1:219:42:219:47 | input | semmle.label | input |
| test.ps1:221:33:221:38 | input | semmle.label | input |
subpaths
#select
| test.ps1:4:23:4:52 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:4:23:4:52 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to Invoke-Expression |
| test.ps1:10:9:10:38 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:10:9:10:38 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to Invoke-Expression |
| test.ps1:16:50:16:79 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:16:50:16:79 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to InvokeScript |
| test.ps1:22:41:22:70 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:22:41:22:70 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to CreateNestedPipeline |
| test.ps1:22:41:22:70 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:22:41:22:70 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to Invoke |
| test.ps1:22:60:22:69 | UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:22:60:22:69 | UserInput | Possible injection path from user input to dangerous call to Invoke |
| test.ps1:28:38:28:67 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:28:38:28:67 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to AddScript |
| test.ps1:28:38:28:67 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:28:38:28:67 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to Invoke |
| test.ps1:28:57:28:66 | UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:28:57:28:66 | UserInput | Possible injection path from user input to dangerous call to Invoke |
| test.ps1:34:14:34:46 | public class Foo { $UserInput } | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:34:14:34:46 | public class Foo { $UserInput } | Possible injection path from user input to dangerous call to Invoke-Expression |
| test.ps1:40:30:40:62 | public class Foo { $UserInput } | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:40:30:40:62 | public class Foo { $UserInput } | Possible injection path from user input to dangerous call to Invoke-Expression |
| test.ps1:48:30:48:34 | code | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:48:30:48:34 | code | Possible injection path from user input to dangerous call to Invoke-Expression |
| test.ps1:75:25:75:54 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:75:25:75:54 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to Powershell |
| test.ps1:82:16:82:45 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:82:16:82:45 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to Powershell |
| test.ps1:89:12:89:28 | ping $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:89:12:89:28 | ping $UserInput | Possible injection path from user input to dangerous call to Cmd |
| test.ps1:106:33:106:62 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:106:33:106:62 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to CreateScriptBlock |
| test.ps1:116:58:116:87 | Get-Process -Name $UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:116:58:116:87 | Get-Process -Name $UserInput | Possible injection path from user input to dangerous call to NewScriptBlock |
| test.ps1:124:34:124:43 | UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:124:34:124:43 | UserInput | Possible injection path from user input to dangerous call to ForEach-Object |
| test.ps1:131:28:131:37 | UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:131:28:131:37 | UserInput | Possible injection path from user input to dangerous call to Invoke |
| test.ps1:138:28:138:37 | UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:138:28:138:37 | UserInput | Possible injection path from user input to dangerous call to Invoke |
| test.ps1:168:50:168:59 | UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:168:50:168:59 | UserInput | Possible injection path from user input to dangerous call to ExpandString |
| test.ps1:176:63:176:72 | UserInput | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:176:63:176:72 | UserInput | Possible injection path from user input to dangerous call to ExpandString |
| test.ps1:192:23:192:54 | Get-Process -Name "$escaped" | test.ps1:195:10:195:32 | Call to Read-Host | test.ps1:192:23:192:54 | Get-Process -Name "$escaped" | Possible injection path from user input to dangerous call to Invoke-Expression |
